sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -513,14 +515,17 @@ public virtual async Task<Response<TableItem>> CreateIfNotExistsAsync(Cancellati
             scope.Start();
             try
             {
-                var context = new RequestContext { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow };
+                RequestContext context = cancellationToken == default ?
+                    TablesRequestContext.AddEntity :
+                    new TablesRequestContext(TablesRequestContext.Conflict) { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow };
+
                 var response = await _tableOperations.CreateAsync(
                     RequestContent.Create(new TableProperties { TableName = Name }),
                     TableConstants.Odata.MinimalMetadata,
                     TableConstants.ReturnNoContent,
                     context).ConfigureAwait(false);
 
-                if (response.IsError)
+                if (response.IsError || response.Status == (int)HttpStatusCode.Conflict)
```

Comment:
Should we consider using the `public static` member from the request context here and avoid the cast?

Reply:
If you are referring to the static on TablesRequestContext, that one is an `int[] { (int)HttoStatusCode.Conflict }` - I suppose we could index into that array, but I don't think the cast is very expensive.

Reply:
ahh... yeah.  My bad, that's what I was thinking of but forgot it was an array.   No concerns on the cast, just thought we had an easily available alias for it already.